### ISSUE TITLE
Run `needs` jobs only one time

### ISSUE THREAD

**Issue by Aohzan** [firecow/gitlab-ci-local#821]
**Is your feature request related to a problem? Please describe.**
Hello, is it possible, when I specify the `--needs` option, to run needed job only the first time and skip them after ?

**Describe the solution you'd like**
Something like `gitlab-ci-local MyJob --needs-if-missing`

**Additional context**
I have a pipeline, with a Lint job that need to install dependencies, but I don't want to download them each time, because this lasts for 1 minute

thank you

**Comment by firecow** [comment]
Well, its possible, but i would consider it extremely nice-to-have, sorry 😁

**Comment by Aohzan** [comment]
Yes but in a classic pipeline when we need to run a job again it just use what was exported by previous job, we don't need to run all needs
in my case, I want to add the lint job in the pre-commit hook of my repo but I don't want users lost 1 minute for each commit, without asking them to run needs before the first commit

**Comment by firecow** [comment]
Won't `gitlab-ci-local lint-job` do just that?

**Comment by Aohzan** [comment]
yes I could check if `.gitlab-ci-local/artifacts/my_needed_job` exists, it will ask a custom command in the pre-commit config but I can do that
otherwise `--ensure-needs` is a good name :+1:

**Comment by Aohzan** [comment]
Hello!
Still interested by the feature 😄
Thank you

**Comment by Aohzan** [comment]
Hello @firecow 
Do you think the feature can be added in a near future ?

**Comment by firecow** [comment]
I can't give any estimates.
I only work on this project when i got time.

**Comment by firecow** [comment]
I can't give any estimates.
I only work on this project when i got time.

**Comment by Aohzan** [comment]
I understand, thank you

**Comment by firecow** [comment]
We should call the option `--ensure-artifacts` shouldn't we?

**Comment by Aohzan** [comment]
as long as needed jobs are included it's good for me
